src/NativeBridge/PythonInterop.h

```diff
@@ -80,29 +88,105 @@ inline PythonObject<T>::PythonObject(const int& kind, size_t numRows, size_t num
 {
     _numRows = numRows;
     _numCols = numCols;
+}
+
+
+/*
+ * PythonObject based class which
+ * handles the single column case.
+ */
+template <class T>
+class PythonObjectSingle : public PythonObject<T>
+{
+protected:
+    std::vector<T>* _pData;
+
+public:
+    PythonObjectSingle(const int& kind, size_t numRows = 1, size_t numCols = 1);
+    virtual ~PythonObjectSingle();
+    virtual void SetAt(size_t nRow, size_t nCol, const T& value);
+    virtual void AddToDict(bp::dict& dict, const std::string& name, const std::vector<std::string>* keyNames);
+    const std::vector<T>* GetData() const { return _pData; }
+};
 
+template <class T>
+inline PythonObjectSingle<T>::PythonObjectSingle(const int& kind, size_t numRows, size_t numCols)
+    : PythonObject<T>(kind, numRows, numCols)
+{
     _pData = new std::vector<T>();
-    if (_numRows > 0)
-        _pData->reserve(_numRows*_numCols);
+    if (numRows > 0)
+        _pData->reserve(numRows*numCols);
```

Comment:
Should this still check num cols since this class should only be handling single valued columns?

Reply:
Fixed in commit 20dda09d19182b8d76859836c2215f3958a4a2eb